persistence/gormdatabase/db.go

```diff
@@ -32,18 +33,21 @@ func createDsn(host, username, password, sslmode, dbname string, port int) strin
 }
 
 // Initialize the database with using the database configuration
-func InitializeDBFromConf(logger *zap.Logger, databaseConfiguration configuration.DatabaseConfiguration) (*gorm.DB, error) {
+func CreateDatabaseConnectionFromConfiguration(logger *zap.Logger, databaseConfiguration configuration.DatabaseConfiguration) (*gorm.DB, error) {
 	dsn := createDsnFromConf(databaseConfiguration)
-	db, err := initializeDBFromDsn(dsn, logger)
-	if err != nil {
-		return nil, err
+	var err error
+	var database *gorm.DB
+	for numberRetry := uint(0); numberRetry < databaseConfiguration.GetRetry(); numberRetry++ {
+		database, err = initializeDBFromDsn(dsn, logger)
+		if err == nil {
+			logger.Sugar().Debugf("Database connection failed with error %q", err.Error())
```

Comment:
not fail in this case

Reply:
or misplace logger

Reply:
sorry i thought I had fixed this in this PR but i was in another 🙃